File: crates/klyron_template/bindings/cpp/builder.cpp
#include "builder.hpp"

namespace klyron::_template {

TemplateBuilder::TemplateBuilder() : config_{} {}

TemplateBuilder& TemplateBuilder::set_config(const TemplateConfig& cfg) {
    config_ = cfg;
    return *this;
}

TemplateConfig TemplateBuilder::config() const {
    return config_;
}

}

File: crates/klyron_template/bindings/rust/builder.rs
//! Builder for klyron_template
use crate::types::TemplateConfig;

pub struct TemplateBuilder {
    config: TemplateConfig,
}

impl TemplateBuilder {
    pub fn new() -> Self {
        Self { config: TemplateConfig::default() }
    }
    pub fn with_version(mut self, version: &str) -> Self {
        self.config.version = version.to_string();
        self
    }
    pub fn build(self) -> TemplateConfig {
        self.config
    }
}
